osbuild/objectstore.py

```diff
@@ -139,39 +162,22 @@ def mode(self) -> Mode:
     def init(self, base: "Object"):
         """Initialize the object with the base object"""
         self._check_mode(Object.Mode.WRITE)
+        if not self._path:
+            raise AssertionError("need active context")
```

Comment:
`__exit__()` does not reset `_path`, so this assertion is maybe misleading. But maybe use `assert self.active` instead? This would not really require a string anymore, as it should be self-explanatory? And it would actually assert for an active context.

Reply:
Same for the other instances of `if not self._path/_meta` below.

Reply:
Or is the intention to always only check for the value you access (`_path`) rather than the property of the object (`active`)? Fine with me, but we either need to reset the values in `__exit__()` or adjust the assertion-message, right?

Reply:
I had `assert self.active` before but then `mypy` was complaining about accessing the instances variables so I change it. I could maybe do:

```
assert self.active
assert self._path  # for mypy
```

Reply:
Yeah, that's what I ended up doing in `FsCache` as well. The first assert for proper exceptions, the second one for mypy, but it cannot ever trigger.